Doc/library/tokenize.rst

```diff
@@ -278,3 +278,15 @@ The exact token type names can be displayed using the :option:`-e` option:
     4,10-4,11:          RPAR           ')'
     4,11-4,12:          NEWLINE        '\n'
     5,0-5,0:            ENDMARKER      ''
+
+Example of tokenizing from another file::
+
+    import tokenize
+    f = open('example.py', 'rb')
```

Comment:
Using `tokenize.open()` would be better as it automatically detect the encoding of the file:

https://github.com/python/cpython/blob/5d90954856cc1d999dd5d153b807055fde0d53ea/Lib/tokenize.py#L449

Reply:
Also, I agree with Emmanuel that using the existing `hello.py` as an example would make the example easier to follow.

Reply:
I found the [tokenize function](https://github.com/python/cpython/blob/master/Lib/tokenize.py#L404) already call `detect_encoding`.

Reply:
I just realized that `tokenize.open()` returns a text stream, so passing it to `tokenize.tokenize()` wouldn't work.

We can use `tokenize.generate_tokens()` to demonstrate how the `str` API works:

```py
import tokenize

with tokenize.open('hello.py') as f:
    token_gen = tokenize.generate_tokens(f.readline)
    for token in token_gen:
        print(token)
```

Then fallback to your example to show the usage of the `bytes` API:

```py
import tokenize

with open('hello.py', 'rb') as f:
    token_gen = tokenize.tokenize(f.readline)
    for token in token_gen:
        print(token)
```